web/app/scripts/src/directives/form.es.js

```diff
@@ -23,37 +23,41 @@ module.directive('surveysForm', [
             restrict: 'A',
             scope: {
                 formUrl: '=',
-                responseUrl: '='
+                responseUrl: '=',
+                onSuccess: '&',
+                onFailure: '&'
             },
             templateUrl: 'templates/incuna-surveys/form/survey-form.html',
-            link: function ($scope) {
-                $scope.model = {};
-                $scope.$watch('formUrl', (url) => {
+            link: function (scope) {
+                scope.model = {};
+                scope.$watch('formUrl', (url) => {
                     if (url) {
                         API.getForm(url).then(function (structure) {
-                            $scope.fields = FieldsetParser.parseFields(structure);
+                            scope.fields = FieldsetParser.parseFields(structure);
                             // Only set the empty model if the model has not
                             // been set.
-                            if (Object.keys($scope.model).length === 0) {
-                                $scope.model = FieldsetParser.parseFormToModel(structure);
+                            if (Object.keys(scope.model).length === 0) {
+                                scope.model = FieldsetParser.parseFormToModel(structure);
                             }
                         });
                     }
                 });
 
-                $scope.$watch('responseUrl', (url) => {
+                scope.$watch('responseUrl', (url) => {
                     if (url) {
                         API.get(url).then(function (data) {
-                            $scope.model = data;
+                            scope.model = data;
                         });
                     }
                 });
 
-                $scope.submit = function () {
-                    if ($scope.responseUrl) {
+                scope.submit = function () {
+                    if (scope.responseUrl) {
                         // TODO: Handle errors.
```

Comment:
I think we still need to handle errors properly within the form - somehow the field errors (such as this is required) need to be displayed by the fields